web/wp-content/themes/dctx-v2/inc/template-tags.php

```diff
@@ -0,0 +1,540 @@
+			if ( $tags_list ) {
+				/* translators: the post tags */
+				printf( '<span class="tags-links">' . esc_html__( 'Tagged %1$s', 'dctx' ) . '</span>', $tags_list ); // WPCS: XSS OK.
+			}
+		}
+
+		if ( ! is_single() && ! post_password_required() && ( comments_open() || get_comments_number() ) ) {
+			echo '<span class="comments-link">';
+			comments_popup_link( esc_html__( 'Leave a comment', 'dctx' ), esc_html__( '1 Comment', 'dctx' ), esc_html__( '% Comments', 'dctx' ) );
+			echo '</span>';
+		}
+
+		edit_post_link(
+			sprintf(
+				/* translators: %s: Name of current post */
+				esc_html__( 'Edit %s', 'dctx' ),
+				the_title( '<span class="screen-reader-text">"', '"</span>', false )
+			),
+			'<span class="edit-link">',
+			'</span>'
+		);
+	}
+endif;
+
+/**
+ * Display SVG Markup.
+ *
+ * @param array $args The parameters needed to get the SVG.
+ *
+ * @author WDS
+ */
+function dctx_display_svg( $args = array() ) {
+	echo dctx_get_svg( $args ); // WPCS XSS Ok.
+}
+
+/**
+ * Return SVG markup.
+ *
+ * @param array $args The parameters needed to display the SVG.
+ * @author WDS
+ * @return string
+ */
+function dctx_get_svg( $args = array() ) {
+
+	// Make sure $args are an array.
+	if ( empty( $args ) ) {
+		return esc_html__( 'Please define default parameters in the form of an array.', 'dctx' );
+	}
+
+	// Define an icon.
+	if ( false === array_key_exists( 'icon', $args ) ) {
+		return esc_html__( 'Please define an SVG icon filename.', 'dctx' );
+	}
+
+	// Set defaults.
+	$defaults = array(
+		'icon'   => '',
+		'title'  => '',
+		'desc'   => '',
+		'fill'   => '',
+		'height' => '',
+		'width'  => '',
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	// Figure out which title to use.
+	$block_title = ( $args['title'] ) ? $args['title'] : $args['icon'];
+
+	// Generate random IDs for the title and description.
+	$random_number  = wp_rand( 0, 99999 );
+	$block_title_id = 'title-' . sanitize_title( $block_title ) . '-' . $random_number;
+	$desc_id        = 'desc-' . sanitize_title( $block_title ) . '-' . $random_number;
+
+	// Set ARIA.
+	$aria_hidden     = ' aria-hidden="true"';
+	$aria_labelledby = '';
+
+	if ( $args['title'] && $args['desc'] ) {
+		$aria_labelledby = ' aria-labelledby="' . $block_title_id . ' ' . $desc_id . '"';
+		$aria_hidden     = '';
+	}
+
+	// Set SVG parameters.
+	$fill   = ( $args['fill'] ) ? ' fill="' . $args['fill'] . '"' : '';
+	$height = ( $args['height'] ) ? ' height="' . $args['height'] . '"' : '';
+	$width  = ( $args['width'] ) ? ' width="' . $args['width'] . '"' : '';
+
+	// Start a buffer...
+	ob_start();
+	?>
+
+	<svg
+	<?php
+		echo dctx_get_the_content( $height ); // WPCS XSS OK.
+		echo dctx_get_the_content( $width ); // WPCS XSS OK.
+		echo dctx_get_the_content( $fill ); // WPCS XSS OK.
+	?>
+		class="icon icon-<?php echo esc_attr( $args['icon'] ); ?>"
+	<?php
+		echo dctx_get_the_content( $aria_hidden ); // WPCS XSS OK.
```

Comment:
:warning: **Warning**: Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native &quot;phpcs:ignore Standard.Category.SniffName.ErrorCode&quot; annotations instead. Found: // WPCS XSS OK.\n (*WordPress.Security.EscapeOutput.DeprecatedWhitelistCommentFound*).